src/overlays/actors/ovl_Bg_Mori_Hashira4/z_bg_mori_hashira4.c

```diff
@@ -20,27 +27,142 @@ const ActorInit Bg_Mori_Hashira4_InitVars = {
     (ActorFunc)BgMoriHashira4_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashira4/func_808A3060.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashira4/func_808A3068.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 700, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern ColHeader D_06001AF8;
+extern ColHeader D_060089E0;
+
+static Gfx* displayLists[] = { 0x06001300, 0x06008840 };
+
+static s16 unkTimer; //seems to be unused
+
+void BgMoriHashira4_SetupAction(BgMoriHashira4* this, BgMoriHashira4ActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgMoriHashira4_InitDynaPoly(BgMoriHashira4* this, GlobalContext* globalCtx, ColHeader* collision, s32 moveFlag) {
+    s32 pad;
+    ColHeader* colHeader;
+    s32 pad2;
+
+    colHeader = NULL;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(collision, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_mori_hashira4.c", 155,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgMoriHashira4_Init(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriHashira4* this = THIS;
+
+    this->switchFlag = (this->dyna.actor.params >> 8) & 0x3F;
+    this->dyna.actor.params &= 0xFF;
+
+    if (this->dyna.actor.params == 0) {
+        BgMoriHashira4_InitDynaPoly(this, globalCtx2, &D_06001AF8, DPM_UNK3);
+    } else {
+        BgMoriHashira4_InitDynaPoly(this, globalCtx2, &D_060089E0, DPM_UNK);
+    }
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->objBankIndex = Object_GetIndex(&globalCtx2->objectCtx, OBJECT_MORI_TEX);
```

Comment:
In order to prevent additional objects from being confused with the actual object for the actor (`actor.objBankIndex`), we try to use explicit variables/functions names for them.
```suggestion
    this->moriTexObjIndex = Object_GetIndex(&globalCtx2->objectCtx, OBJECT_MORI_TEX);
```